pkg/volume/azure_file/azure_file.go

```diff
@@ -104,6 +105,13 @@ func (plugin *azureFilePlugin) newMounterInternal(spec *volume.Spec, pod *v1.Pod
 	if err != nil {
 		return nil, err
 	}
+	nameSpace := pod.Namespace
+	if spec.PersistentVolume != nil {
+		ns, err := getSecretNamespace(plugin.host.GetKubeClient(), spec.PersistentVolume)
```

Comment:
create/delete and attach/detach are done by controllers living outside the nodes, so it is more reasonable to grant broader permission to them, and/or have them resolve storage class info.

mount/unmount is done by arbitrary nodes, so segmenting secret access is far more important